src/services/studio/studio.service.ts

```diff
@@ -0,0 +1,49 @@
+import {PackageExportTransport} from "../../interfaces/package-export-transport";
+import {packageApi} from "../../api/package-api";
+import {PackageManagerVariableType, PackageWithVariableAssignments} from "../../interfaces/package-manager.interfaces";
+import {dataModelService} from "../package-manager/datamodel-service";
+
+class StudioService {
+
+    public async getExportPackagesWithStudioData(packagesToExport: PackageExportTransport[], withDependencies: boolean) {
+        const studioPackagesWithDataModels = await packageApi.findAllPackagesWithVariableAssignments(PackageManagerVariableType.DATA_MODEL);
+
+        await studioService.setSpaceIdForStudioPackages(packagesToExport, studioPackagesWithDataModels);
+
+        if (withDependencies) {
+            await studioService.setDataModelsForStudioPackages(packagesToExport, studioPackagesWithDataModels);
+        }
+
+        return packagesToExport;
+    }
+
+    private async setSpaceIdForStudioPackages(packages: PackageExportTransport[], studioPackages: PackageWithVariableAssignments[]): Promise<void> {
+        const packageByKey = new Map<string, PackageExportTransport>();
```

Comment:
from personal experience using an object (Record<string. PackageExportTransport) as a Map tends to be easier as well as more performant if the Map keys are simply keys

Reply:
Can you explain a bit what do you mean by 'if the Map keys are simply keys'? I thought using a map here would be better since we are doing key searches.

Reply:
an object is basically a map right where the key is a string and the value is anything for example:
```ts
const obj = {
  firstProperty: 5,
  'second-property': true
  thirdProperty: {
    a: 'b'
  }
}
```
And the common map functions can be represented as:
```ts
obj['firstProperty']; (map.get())
'second-property' in obj; (map.has())
obj['thirdProperty'] = {}
```

Reply:
I don't think the performance gains would be significant in our case and I find it more readable and consistent if we stick to maps since they are used throughout the CLI.